pkg/theia/commands/policy_recommendation_status.go

```diff
@@ -67,73 +68,118 @@ $ theia policy-recommendation status e998433e-accb-4888-9fc8-06563f073e86 --use-
+			if err != nil {
+				return err
+			}
+			if state == "" {
+				state = "NEW"
+			}
+			if state == "RUNNING" {
+				var endpoint string
+				service := fmt.Sprintf("pr-%s-ui-svc", recoID)
+				if useClusterIP {
+					serviceIP, servicePort, err := GetServiceAddr(clientset, service)
+					if err != nil {
+						klog.V(2).ErrorS(err, "error when getting the progress of the job, cannot get Spark Monitor Service address")
+					} else {
+						endpoint = fmt.Sprintf("tcp://%s:%d", serviceIP, servicePort)
+					}
 				} else {
-					endpoint = fmt.Sprintf("tcp://%s:%d", serviceIP, servicePort)
+					servicePort := 4040
+					listenAddress := "localhost"
+					listenPort := 4040
+					pf, err := StartPortForward(kubeconfig, service, servicePort, listenAddress, listenPort)
+					if err != nil {
+						klog.V(2).ErrorS(err, "error when getting the progress of the job, cannot forward port")
+					} else {
+						endpoint = fmt.Sprintf("http://%s:%d", listenAddress, listenPort)
+						defer pf.Stop()
+					}
 				}
-			} else {
-				servicePort := 4040
-				listenAddress := "localhost"
-				listenPort := 4040
-				pf, err := StartPortForward(kubeconfig, service, servicePort, listenAddress, listenPort)
-				if err != nil {
-					klog.V(2).ErrorS(err, "error when getting the progress of the job, cannot forward port")
-				} else {
-					endpoint = fmt.Sprintf("http://%s:%d", listenAddress, listenPort)
-					defer pf.Stop()
+				// Check the working progress of running recommendation job
+				if endpoint != "" {
+					stateProgress, err := getPolicyRecommendationProgress(endpoint)
+					if err != nil {
+						klog.V(2).ErrorS(err, "failed to get the progress of the job")
+					}
+					state += stateProgress
 				}
 			}
-			// Check the working progress of running recommendation job
-			if endpoint != "" {
-				stateProgress, err := getPolicyRecommendationProgress(endpoint)
-				if err != nil {
-					klog.V(2).ErrorS(err, "failed to get the progress of the job")
-				}
-				state += stateProgress
+			errorMessage, err = getPolicyRecommendationErrorMsg(clientset, recoID)
+			if err != nil {
+				return err
 			}
+		} else {
+			state = "COMPLETED"
 		}
 		fmt.Printf("Status of this policy recommendation job is %s\n", state)
+		if errorMessage != "" {
+			fmt.Printf("Error message: %s\n", errorMessage)
+		}
 		return nil
 	},
 }
 
-func getPolicyRecommendationStatus(clientset kubernetes.Interface, recoID string) (string, error) {
-	sparkApplication := &sparkv1.SparkApplication{}
-	err := clientset.CoreV1().RESTClient().
+func getSparkAppByRecommendationID(clientset kubernetes.Interface, id string) (sparkApp sparkv1.SparkApplication, err error) {
+	err = clientset.CoreV1().RESTClient().
 		Get().
 		AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
 		Namespace(flowVisibilityNS).
 		Resource("sparkapplications").
-		Name("pr-" + recoID).
+		Name("pr-" + id).
 		Do(context.TODO()).
-		Into(sparkApplication)
+		Into(&sparkApp)
```

Comment:
Apology if I missed something... Where does the `sparkApp` come from? I didn't see its initialization.

Reply:
In the return value list of this function.